frontend/src/components/SamplesList.tsx

```diff
@@ -317,11 +324,14 @@ export default function SamplesList({
             });
             return buildTsvString(
               data.dashboardSamples,
-              columnDefs,
+              columnDefsForExport,
               gridRef.current?.columnApi?.getAllGridColumns()
             );
           }}
-          onComplete={() => setShowDownloadModal(false)}
+          onComplete={() => {
+            setShowDownloadModal(false);
+            setColumnDefsForExport(columnDefs);
+          }}
```

Comment:
Added a new state `columnDefsForExport` that tracks what columns show up in the export.

By default, it exports `columnDefs` -- the columns that are shown to users. 

When users click on "Generate Phenotype files for dbGaP", we temporarily set `columnDefsForExport` to the new [`DbGapPhenotypeColumns`](https://github.com/mskcc/smile-dashboard/pull/188/files#diff-db966dd79103f32fddb49dcefcfba78d4325506ce364cd0e24779a3521aa7173R500). When the export completes, we revert `columnDefsForExport` back to the default `columnDefs`.